cmd/kubeadm/app/phases/apiconfig/setupmaster.go

```diff
@@ -49,6 +50,8 @@ func findMyself(client *clientset.Clientset) (*v1.Node, error) {
 }
 
 func attemptToUpdateMasterRoleLabelsAndTaints(client *clientset.Clientset) error {
+	validate.WaitForNode(client, false)
```

Comment:
This seems less racy but still racy. What if a non-master node registers before the master node? We will get the same crash. It's also strange that we have a phase depending on another phase. What if you just:
```go
wait.PollInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
    if _, err := client.Nodes().Get(masterNodeName); err != nil {
        return true, nil
    }
    return false, nil
}
```
And leave validate as is? I also don't see value in sharing code with WaitForNode if the change requires adding significantly more lines of code then just adding a PollInfinite inline here.